File: CF_Quick-One/920_div3/C.py
from sys import stdin
def input(): return stdin.readline()[:-1]

def solve():
    N,f, a, b = map(int, input().split())
    Arr = list(map(int, input().split()))
    Arr.insert(0,0)
    
    for i in range(0, N + 1):
        if f <= 0:
            print("NO")
            return
        if i == N:
            print("YES")
            return
        nothing = Arr[i + 1] - Arr[i]
        nothing *= a
        nothing = min(nothing,  b )
        f -= nothing
        
    
    
for _ in range(int(input())):
    solve()

File: CF_Quick-One/920_div3/C.cpp
#include <algorithm>
#include <iostream>
using namespace std;

#define int long long

void solve(int n, int f, int a, int b, int arr[]) {
    // print array
    // for (int i = 0; i < n; i++) cout << arr[i] << " ";

    int sum = 0;
    for (int i = 0; i < n - 1; i++) {
        int t = arr[i + 1] - arr[i];
        int e = a * t;
        sum += min(e, b);
    }

    sum += min(a*arr[0], b);
    // cout << sum << endl;

    if (sum < f)
        printf("Yes\n");
    else
        printf("No\n");
}

int32_t main() {
    // fast io
    ios_base::sync_with_stdio(false);
    cin.tie(NULL);

    int t;
    cin >> t;
    while (t--) {
        int n, f, a, b;
        cin >> n >> f >> a >> b;
        int arr[n];
        for (int i = 0; i < n; i++) cin >> arr[i];
        solve(n, f, a, b, arr);
    }
    return 0;
}